src/HttpAsyncClientDiscovery.php

```diff
@@ -24,7 +24,7 @@ public static function find()
         try {
             $asyncClient = static::findOneByType(HttpAsyncClient::class);
 
-            return new $asyncClient();
+            return static::instantiateClass($asyncClient);
```

Comment:
No, we do not need a try-catch here. We know that instantiateClass could only throw a `ClassInstantiationFailedException`